src/helpers.cpp

```diff
@@ -105,12 +105,12 @@ ada_really_inline bool shorten_path(std::string& path,
   // Let path be url’s path.
   // If url’s scheme is "file", path’s size is 1, and path[0] is a normalized
   // Windows drive letter, then return.
-  if (type == ada::scheme::type::FILE &&
+  if (path.length() && type == ada::scheme::type::FILE &&
       first_delimiter == std::string_view::npos) {
     if (checkers::is_normalized_windows_drive_letter(
-            std::string_view(path.data() + 1, first_delimiter - 1))) {
-      return false;
-    }
+            std::string_view(path.data() + 1, path.length() - 1))) {
```

Comment:
With adding length check to parent if statement, I wonder if this change is still valid. we made sure that npos is a valid not found, not the result of an empty string search.

Reply:
good point - i can get that fixed up!

Reply:
@anonrig the crash still happens if I switch from `path.length() - 1` to `first_delimeter -1` so i do think we might need to keep it or at least something similar 🤔